src/filtering.ts

```diff
@@ -0,0 +1,69 @@
+export type AIResponse = {
+  flagged: boolean
+  reason: string
+}
+
+export interface Message {
+  text: string
+  threadId: string
+  messageId: string
+  parent: string | null
+  flagged?: boolean
+  reason?: string
+}
+
+export interface UserMessage {
+  message: Message
+  timestamp: number
+  username: string
+  address: string
+}
+export async function callAI(prompt: string, userInput: string, url: string, token: string): Promise<AIResponse> {
+  const requestBody = {
+    messages: [
+      {
+        role: 'system',
+        content: prompt,
+      },
+      {
+        role: 'user',
+        content: `Comment to check: ${userInput}`,
+      },
+    ],
+  }
+
+  const controller = new AbortController()
+  const timeoutId = setTimeout(() => controller.abort(), 3000)
+
+  const response = await fetch(url, {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${token}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(requestBody),
+    signal: controller.signal,
+  })
+    .catch((error: any) => {
+      //console.log(error)
+
+      return new Response(null, { status: 408, statusText: 'Request Timeout' })
+    })
+    .finally(() => clearTimeout(timeoutId))
+
+  if (response.status === 408) {
+    return { flagged: true, reason: 'Request timed out' }
+  }
```

Comment:
This effectively means that content will be censored, when there was an error calling an API endpoint.
Do we want this? This makes some sense, because it is safe, on the other hand, we might lose a lot of messages when there is an error with the API.

Reply:
Yes as we discussed in this case the timeout should be logged as an error, but the flag should not be set to true because it is more likely that it happens intermittently and less likely that we get content  in need of moderating at that exact time.

Reply:
fix: https://github.com/Solar-Punk-Ltd/gateway-proxy/commit/56fc7159f21d639ebb5743f414843069f80ba717
This commit also return flagged=false when parsing error occur

Reply:
fix: https://github.com/Solar-Punk-Ltd/gateway-proxy/blob/lb-ai/src/filtering.ts#L57-L58
This commit log out timeout and return flagged=false